Source/ACE/Entity/Player.cs

```diff
@@ -1388,7 +1391,56 @@ public override void Tick(double tickTime)
                 Session.Network.EnqueueSend(new GameMessagePrivateUpdateAttribute2ndLevel(Session, Vital.Mana, Mana.Current));
             }
         }
-        
+
+        public void EquipUnequipItem(uint itemId, uint slot)
+        {
+            UpdateWieldedItem(itemId);
+            ModelData.Clear();
+            AddCharacterBaseModelData(); // Add back in the facial features, hair and skin palette
+            var wieldeditems = GetCurrentlyWieldedItems();
+            var coverage = new List<uint>();
+
+            foreach (var w in wieldeditems)
+            {
+                var wo = w.Value;
+                ClothingTable item;
+                if (wo.ClothingBase != null)
+                    item = ClothingTable.ReadFromDat((uint)wo.ClothingBase);
+                else
+                    item = ClothingTable.ReadFromDat(0x10000002);
+                // If I don't find a match, for now you are pants!
```

Comment:
Maybe change this to some kind of error catcher with a do nothing on the model + chat error like portals do when they aren't wired up, so we don't end up with some mixed results?